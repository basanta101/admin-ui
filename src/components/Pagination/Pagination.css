.paginationWrapper {
  height: 20px;
  width: 200px;
  display: flex;
  justify-content: space-around;
}
