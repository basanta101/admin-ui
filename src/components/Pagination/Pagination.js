import "./Pagination.css";

const Pagination = ({ pagination = {}, handlePageChange = (f) => f }) => {
  const { currentPageNo, totalPages } = pagination;

  // disable btn when it is not required
  const isFirstPageInView = currentPageNo === 1;
  const isLastPageInView = currentPageNo === totalPages;
  
  return (
    <div className="paginationWrapper">
      <button
        onClick={() => handlePageChange(1)}
        disabled={isFirstPageInView}
        data-testid='getFirstPageBtn'
      >
        {"<<"}
      </button>
      <button
        onClick={() => handlePageChange(currentPageNo - 1)}
        disabled={isFirstPageInView}
        data-testid='getPrevPageBtn'
      >
        {"<"}
      </button>
      <span>{currentPageNo}</span>/<span>{totalPages}</span>
      <button
        onClick={() => handlePageChange(currentPageNo + 1)}
        disabled={isLastPageInView}
        data-testid='getNextPageBtn'

      >
        {">"}
      </button>
      <button
        onClick={() => handlePageChange(totalPages)}
        disabled={isLastPageInView}
        data-testid='getLastPageBtn'
      >
        {">>"}
      </button>
    </div>
  );
};

export default Pagination;
