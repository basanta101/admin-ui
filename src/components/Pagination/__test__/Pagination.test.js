import { render, fireEvent, getByTestId } from "@testing-library/react";
import renderer from "react-test-renderer";
import Pagination from "../Pagination";

const setup = () => {
  const pagination = { currentPageNo: 3, totalPages: 5 };
  const handlePageChange = jest.fn();
  const { container } = render(
    <Pagination handlePageChange={handlePageChange} pagination={pagination} />
  );

  return {
    container,
    handlePageChange,
  };
};

test("clicking << takes us to the first page, if currentPage is not first page", () => {
  const { container, handlePageChange } = setup();
  fireEvent.click(getByTestId(container, 'getFirstPageBtn'));
  expect(handlePageChange).toHaveBeenCalledTimes(1);
});

test("clicking < takes us to the previous page, if currentPage is not first page", () => {
  const { container, handlePageChange } = setup();
  fireEvent.click(getByTestId(container, 'getPrevPageBtn'));
  expect(handlePageChange).toHaveBeenCalledTimes(1);
});

test("clicking > takes us to the next page, if currentPage is not last page", () => {
  const { container, handlePageChange } = setup();
  fireEvent.click(getByTestId(container, 'getNextPageBtn'));
  expect(handlePageChange).toHaveBeenCalledTimes(1);
});

test("clicking >> takes us to the next page, if currentPage is not last page", () => {
  const { container, handlePageChange } = setup();
  fireEvent.click(getByTestId(container, 'getLastPageBtn'));
  expect(handlePageChange).toHaveBeenCalledTimes(1);
});

it("renders a snapshot of Pagination Component", () => {
  const tree = renderer.create(<Pagination />).toJSON();
  expect(tree).toMatchSnapshot();
});
