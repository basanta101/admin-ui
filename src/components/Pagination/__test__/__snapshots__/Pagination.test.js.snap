// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of Pagination Component 1`] = `
<div
  className="paginationWrapper"
>
  <button
    data-testid="getFirstPageBtn"
    disabled={false}
    onClick={[Function]}
  >
    &lt;&lt;
  </button>
  <button
    data-testid="getPrevPageBtn"
    disabled={false}
    onClick={[Function]}
  >
    &lt;
  </button>
  <span />
  /
  <span />
  <button
    data-testid="getNextPageBtn"
    disabled={true}
    onClick={[Function]}
  >
    &gt;
  </button>
  <button
    data-testid="getLastPageBtn"
    disabled={true}
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;
